package/MDAnalysis/due.py

```diff
@@ -0,0 +1,99 @@
+# emacs: at the end of the file
+# ex: set sts=4 ts=4 sw=4 et:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### #
+"""Stub file for a guaranteed safe import of duecredit constructs:  if duecredit
+is not available.
+
+To use it, place it into your project codebase to be imported, e.g. copy as
+
+    cp stub.py /path/tomodule/module/due.py
+
+Note that it might be better to avoid naming it duecredit.py to avoid shadowing
+installed duecredit.
+
+Then use in your code as
+
+    from .due import due, Doi, BibTeX
+
+See  https://github.com/duecredit/duecredit/blob/master/README.md for examples.
+
+Origin:     Originally a part of the duecredit
+Copyright:  2015-2016  DueCredit developers
+License:    BSD-2
+
+Modified for MDAnalysis to avoid calls to fork which raises cryptic
+warning under MPI(see PR #1794 for rationale)
+
+"""
+
+__version__ = '0.0.5'
+
+
+class InactiveDueCreditCollector(object):
+    """Just a stub at the Collector which would not do anything"""
+    def _donothing(self, *args, **kwargs):
+        """Perform no good and no bad"""
+        pass
+
+    def dcite(self, *args, **kwargs):
+        """If I could cite I would"""
+        def nondecorating_decorator(func):
+            return func
+        return nondecorating_decorator
+
+    cite = load = add = _donothing
+
+    def __repr__(self):
+        return self.__class__.__name__ + '()'
+
+
+def _donothing_func(*args, **kwargs):
+    """Perform no good and no bad"""
+    pass
+
+
+try:
+    # Avoid call of fork inside duecredit; see
+    # https://github.com/MDAnalysis/mdanalysis/pull/1822#issuecomment-373009050
+    import sys
+    import os
+    if sys.version_info >= (3, 7):
+        import duecredit
+    else:
+        from mock import patch
+        if sys.version_info <= (2, ):
+            from contextlib import nested
+            with nested(patch('os.fork'), patch('os.popen')) \
+                 as (os_dot_fork, os_dot_popen):
+                import duecredit
+        else:
+            with patch('os.fork') as os_dot_fork, patch('os.popen') as os_dot_popen:
```

Comment:
Is this doing the same as `nested()` or do we need to use [ExitStack](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack)?

Reply:
Looks good, afaik that is the correct Python 3 way of "nesting" context managers.